XenAdmin/SettingsPanels/SnmpEditPage.cs

```diff
@@ -0,0 +1,278 @@
+                }
+                if (!SupportV2cCheckBox.Checked && !SupportV3CheckBox.Checked)
+                {
+                    _invalidParamToolTip.Tag = SupportV2cCheckBox;
+                    _invalidParamToolTip.ToolTipTitle = Messages.SNMP_ALLOW_CHOOSE_TITLE;
+                    return false;
+                }
+                if (SupportV2cCheckBox.Checked)
+                {
+                    var communityStr = CommunityTextBox.Text;
+                    if (!RegexCommon.Match(communityStr.Trim()).Success)
+                    {
+                        _invalidParamToolTip.Tag = CommunityTextBox;
+                        _invalidParamToolTip.ToolTipTitle = Messages.SNMP_ALLOW_COMMUNITY_TITLE;
+                        _invalidParamToolTipText = Messages.SNMP_ALLOW_COMMUNITY_TEXT;
+                        return false;
+                    }
+                }
+                if (SupportV3CheckBox.Checked)
+                {
+                    var securityStr = UserNameTextBox.Text;
+                    var authPassStr = AuthenticationPasswordLabelTextBox.Text;
+                    var privacyPassStr = PrivacyPasswordTextBox.Text;
+                    if (!RegexCommon.Match(securityStr.Trim()).Success)
+                    {
+                        _invalidParamToolTip.Tag = UserNameTextBox;
+                        _invalidParamToolTip.ToolTipTitle = Messages.SNMP_ALLOW_USER_TITLE;
+                        _invalidParamToolTipText = Messages.SNMP_ALLOW_COMMUNITY_TEXT;
+                        return false;
+                    }
+
+                    if (!RegexEncryptTextBox.Match(authPassStr.Trim()).Success)
+                    {
+                        _invalidParamToolTip.Tag = AuthenticationPasswordLabelTextBox;
+                        _invalidParamToolTip.ToolTipTitle = Messages.SNMP_ALLOW_AUTH_TITLE;
+                        _invalidParamToolTipText = Messages.SNMP_ALLOW_AUTH_TEXT;
+                        return false;
+                    }
+
+                    if (!RegexEncryptTextBox.Match(privacyPassStr.Trim()).Success)
+                    {
+                        _invalidParamToolTip.Tag = PrivacyPasswordTextBox;
+                        _invalidParamToolTip.ToolTipTitle = Messages.SNMP_ALLOW_PRIVACY_TITLE;
+                        _invalidParamToolTipText = Messages.SNMP_ALLOW_AUTH_TEXT;
+                        return false;
+                    }
+                }
+                return true;
+            }
+        }
+        public SnmpEditPage()
+        {
+            InitializeComponent();
+            Text = Messages.SNMP;
+        }
+        public void Cleanup()
+        {
+            _invalidParamToolTip.Dispose();
+        }
+        public void ShowLocalValidationMessages()
+        {
+            if (_invalidParamToolTip.Tag is Control ctrl)
+            {
+                _invalidParamToolTip.Hide(ctrl);
+                if (!_invalidParamToolTip.ToolTipTitle.Equals(""))

```

Comment:
A better way to check strings is `string.IsNullOrEmpty(_invalidParamToolTip.ToolTipTitle)`